src/agent/src/watcher.rs

```diff
@@ -253,23 +256,23 @@ impl SandboxStorages {
     }
 
     async fn check(&mut self, logger: &Logger) -> Result<()> {
-        for entry in self.0.iter_mut() {
-            if !entry.watch {
-                continue;
-            }
-
-            match entry.scan(logger).await {
-                Err(error) => {
+        for entry in self.0.iter_mut().filter(|e| e.watch) {
+            entry
+                .clone()
```

Comment:
Is this `clone` needed because of borrow checker?
I guess there are other ways to avoid this, like:

```rust
self.0.iter_mut().filter_map(|e| {
  if !e.watch {
     return None
  }
  //remaining logic.
   ...
  Some(xyz) 
})
```

Reply:
I think `clone` is not needed. This compiled: https://github.com/egernst/kata-containers/pull/3#discussion_r657509477

Reply:
The clone() is needed because `entry` is borrowed first and then moved into the closure. But `or_else` is using the initial borrow. I'm not sure how `filter_map` or `for_each` would change that situation?